logging.Logger.addFilter

```
porcupine/plugins/langserver.py:789: error: Argument 1 to "addFilter" of "Filterer" has incompatible type "HackFilter"; expected "Union[Filter, Callable[[LogRecord], int]]"
```

`HackFilter` is a class that has a `filter()` method but doesn't inherit from anything. This should work, according to docs: "You don’t actually need to subclass Filter: you can pass any instance which has a filter method with the same semantics." https://docs.python.org/3/library/logging.html

`Callable[[LogRecord], int]` seems wrong. The imlpementation basically does `self.filters.append(filter)`. Does it really handle callables and objects with a `.filter()` method everywhere where it uses `self.filters`?

--- a/stdlib/logging/__init__.pyi
+++ b/stdlib/logging/__init__.pyi
@@ -64,7 +64,7 @@ if sys.version_info >= (3, 11):
 _SysExcInfoType: TypeAlias = Union[tuple[type[BaseException], BaseException, TracebackType | None], tuple[None, None, None]]
 _ExcInfoType: TypeAlias = None | bool | _SysExcInfoType | BaseException
 _ArgsType: TypeAlias = tuple[object, ...] | Mapping[str, object]
-_FilterType: TypeAlias = Filter | Callable[[LogRecord], int]
+_FilterType: TypeAlias = Filter | Callable[[LogRecord], bool]
 _Level: TypeAlias = int | str
 _FormatStyle: TypeAlias = Literal["%", "{", "$"]
 

